lib/screens/user_liked_songs_page.dart

```diff
@@ -116,22 +116,26 @@ class _UserLikedSongsState extends State<UserLikedSongs> {
               ],
             ),
             const Padding(padding: EdgeInsets.only(top: 20)),
-            ListView.builder(
-              shrinkWrap: true,
-              physics: const BouncingScrollPhysics(),
-              addAutomaticKeepAlives: false,
-              addRepaintBoundaries: false,
-              itemCount: userLikedSongsList.length,
-              itemBuilder: (BuildContext context, int index) {
-                return Padding(
-                  padding: const EdgeInsets.only(top: 5, bottom: 5),
-                  child: SongBar(
-                    userLikedSongsList[index],
-                    true,
-                  ),
-                );
-              },
-            )
+            ValueListenableBuilder(
+                valueListenable: currentLikedSongsLength,
+                builder: (_, value, __) {
+                  return ListView.builder(
+                    shrinkWrap: true,
+                    physics: const BouncingScrollPhysics(),
+                    addAutomaticKeepAlives: false,
+                    addRepaintBoundaries: false,
+                    itemCount: userLikedSongsList.length,
```

Comment:
Because when the re-rendering happens, the single item in the list doesn't get the current index of the song, and the app crashes. I tried keeping currentLikedSongsLength and that is what happened to me

Reply:
> Because when the re-rendering happens, the single item in the list doesn't get the current index of the song, and the app crashes. I tried keeping currentLikedSongsLength and that is what happened to me

Then it's okay, let it be how it is.